src/app/admin/components/search-bar/search-bar.component.ts

```diff
@@ -1,15 +1,29 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, AfterViewInit, Input, Output, ViewChild, ElementRef, EventEmitter } from '@angular/core';
+import { fromEvent, Subject } from 'rxjs';
+import { map, debounceTime, takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'clark-admin-search-bar',
   templateUrl: './search-bar.component.html',
   styleUrls: ['./search-bar.component.scss']
 })
-export class SearchBarComponent implements OnInit {
+export class SearchBarComponent implements AfterViewInit {
+  @Input() placeholder: string;
+  @Output() userInput = new EventEmitter<string>();
+  @ViewChild('searchInput', { read: ElementRef })
+  searchInput: ElementRef;
+  destroyed$: Subject<void> = new Subject();
 
   constructor() { }
 
-  ngOnInit() {
+  ngAfterViewInit() {
+    fromEvent(this.searchInput.nativeElement, 'input').pipe(
+      map(x => x['currentTarget'].value),
+      debounceTime(650),
```

Comment:
Should `debounceTime` happen first so that we don't map unless the timer has passed?

Reply:
@nrwinner any thought on this? I tried playing around with this and found that the currentTarget property is set to null after debounce.

Reply:
I'm not sure, I wouldn't think currentTarget would be null after the debounce unless the event is somehow expiring?